### ISSUE TITLE
Auth with sites using Caddy

### ISSUE THREAD

**Issue by abnerjacobsen** [filebrowser/filebrowser#204]
I'm testing filemanager with caddy and would like to know how to generate an authentication database to be used by filemanager before the caddy is started where I can associate a website hosted on Caddy with one or more accounts that are allowed to use filemanager to edit the site? Is this possible ?

**Comment by hacdias** [comment]
You can create the users after starting Caddy. The users will be saved to the database and persist across reboots and such. I recommend you to set the 'database' option to a specific path.

**Comment by abnerjacobsen** [comment]
@hacdias In my case this method does not work. My users exist in a central database and I get the list of them - user, password and permissions - via an API call and with this information I create a Caddyfile on demand.

I did another test with the following configuration in Caddyfile:

```
test2.sites.daspanel.site:443 {
    errors stdout
    log / stdout "[DASPANEL-HTTP-ACCESSLOG] {>Referer} {remote} {when} {method} {uri} {proto} {status} {size} {>User-Agent} {latency} {request}" 
    root /opt/daspanel/data/civmw76wg000001p2dwqxpvet/content/cj62tq3ti00005rnrilfs850o/v/2017-08-08-000431-062486
    gzip
    tls self_signed
    header / {
        X-Frame-Options        SAMEORIGIN
        X-Content-Type-Options nosniff
        X-XSS-Protection       "1; mode=block"
        Referrer-Policy        no-referrer-when-downgrade
        Strict-Transport-Security "max-age=31536000; includeSubdomains; preload"
    }
   basicauth /_das/fm admin@daspanel.site aI81REN4a1H2gy3b
   filemanager /_das/fm /opt/daspanel/data/civmw76wg000001p2dwqxpvet/content/cj62tq3ti00005rnrilfs850o/v/2017-08-08-000431-062486 {
        no_auth
        allow_commands true
        allow_edit true
        allow_new true
        allow_publish true
        commands git
    }
}
```

As expected, Caddy asks for authentication but after informing it correctly, the login dialog reappears again and again, no matter how many times I enter the login data. In addition, if I click the Cancel button in the Cady login dialog (Basicauth) the Filemanager login screen appears.

Is it possible to use Caddy's authentication middleware in conjunction with Filemanager?

What's wrong with the above example of Caddy setup?

Thanks

**Comment by hacdias** [comment]
> Is it possible to use Caddy's authentication middleware in conjunction with Filemanager?

Yes, it is.

> In addition, if I click the Cancel button in the Cady login dialog (Basicauth) the Filemanager login screen appears.

This is strange since we're using `no_auth`. It shouldn't happen. I'll investigate.

**Comment by hacdias** [comment]
Fixed with 1.2.6 😄 You can download the new version from caddyserver.com!

**Comment by abnerjacobsen** [comment]
Thanks. It's worked fine.
